AmahiAnywhere/AmahiAnywhere/Presentation/Login/LoginViewController.swift

```diff
@@ -41,27 +56,47 @@ class LoginViewController: BaseUIViewController {
         return true
     }
     
+    @IBAction func passwordFieldChanged(_ sender: Any) {
+        showHideButton.isHidden = !passwordInputField.hasText
+    }
+    
     @IBAction func userClickForgotPassword(_ sender: UIButton) {
         UIApplication.shared.open(NSURL(string:"https://www.amahi.org/forgot")! as URL)
     }
     
     @IBAction func userclickSignIn(_ sender: Any) {
+        if checkInputsValid(){
+            presenter.login(username: usernameInputField.text!,
+                            password: passwordInputField.text!)
+        }
+    }
+    
+    func checkInputsValid() -> Bool{
+        var isValid = true
```

Comment:
This method is doing more than what it should be doing. It is not just checking if it is valid but setting error texts. I honestly doing see anything wrong in the way the validation was done earlier.

Reply:
Yes you're right, I'll make a separate function for displaying the error messages. The problem with the previous implementation was that it only showed the error message for the username field when both fields were empty. It should show the error message for both fields.

Reply:
This seems okay but can be made better though.